docs/serialization.rst

```diff
@@ -46,35 +46,256 @@ When Hazelcast Python client serializes an object:
+  changing the types of fields
+- Platform and language independent
+- Supports partial deserialization of fields, without deserializing the whole
+  objects during queries or indexing
+
+Hazelcast achieves these features by having a well-known schema of objects and
+replicating them across the cluster which enables members and clients to fetch
+schemas they don’t have in their local registries. Each serialized object
+carries just a schema identifier and relies on the schema distribution service
+or configuration to match identifiers with the actual schema. Once the schemas
+are fetched, they are cached locally on the members and clients so that the
+next operations that use the schema do not incur extra costs.
+
+Schemas help Hazelcast to identify the locations of the fields on the
+serialized binary data. With this information, Hazelcast can deserialize
+individual fields of the data, without reading the whole binary. This results
+in a better query and indexing performance.
+
+Schemas can evolve freely by adding or removing fields. Even, the types of the
+fields can be changed. Multiple versions of the schema may live in the same
+cluster and both the old and new readers may read the compatible parts of the
+data. This feature is especially useful in rolling upgrade scenarios.
+
+The Compact serialization does not require any changes in the user classes as
+it doesn’t need a class to extend another class. Serializers might be
+implemented and registered separately from the classes.
+
+The underlying format of the compact serialized objects is platform and
+language independent.
+
+Using Compact Serialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Compact serialization can be used by writing a serializer that extends the
+:class:`CompactSerializer <hazelcast.serialization.api.CompactSerializer>`
+for a class and registering it in the client configuration.
+
+For example, assume that you have the following ``Employee`` class:
+
+.. code:: python
+
+    class Employee:
+        def __init__(self, name: str, age: int):
+            self.name = name
+            self.age = age
+
+
+Then, a serializer for it can be implemented as below:
+
+.. code:: python
+
+    from hazelcast.serialization.api import CompactSerializer, CompactWriter, CompactReader
+
+    class EmployeeSerializer(CompactSerializer[Employee]):
+        def read(self, reader: CompactReader) -> Employee:
+            name = reader.read_string("name")
+            age = reader.read_int32("age")
+            return Employee(name, age)
+
+        def write(self, writer: CompactWriter, obj: Employee) -> None:
+            writer.write_string("name", obj.name)
+            writer.write_int32("age", obj.age)
+
+        def get_type_name(self) -> str:
+            return "employee"
+
+The last step is to register the serializer in the client configuration.
+
+.. code:: python
+
+    client = HazelcastClient(
+        compact_serializers={
+            Employee: EmployeeSerializer(),
```

Comment:
I thought we decided to change this.

Reply:
I thought we said we would stick to this in our discussion after seeing Sancar's message

Reply:
I'll resolve this so we can merge the PR. I'll send a separate PR to sync this with the latest decision.